tests/integration/test_reload_clusters_config/test.py

```diff
@@ -0,0 +1,235 @@
+                    <port>9000</port>
+                </replica>
+            </shard>
+        </test_cluster2>
+    </remote_servers>
+</yandex>
+'''
+
+test_config2 = '''
+<yandex>
+    <remote_servers>
+        <test_cluster>
+            <shard>
+                <internal_replication>true</internal_replication>
+                <replica>
+                    <host>node_1</host>
+                    <port>9000</port>
+                </replica>
+                <replica>
+                    <host>node_2</host>
+                    <port>9000</port>
+                </replica>
+            </shard>
+        </test_cluster>
+    </remote_servers>
+</yandex>
+'''
+
+test_config3 = '''
+<yandex>
+    <remote_servers>
+        <test_cluster>
+            <shard>
+                <internal_replication>true</internal_replication>
+                <replica>
+                    <host>node_1</host>
+                    <port>9000</port>
+                </replica>
+                <replica>
+                    <host>node_2</host>
+                    <port>9000</port>
+                </replica>
+            </shard>
+        </test_cluster>
+        <test_cluster2>
+            <shard>
+                <internal_replication>true</internal_replication>
+                <replica>
+                    <host>node_1</host>
+                    <port>9000</port>
+                </replica>
+                <replica>
+                    <host>node_2</host>
+                    <port>9000</port>
+                </replica>
+            </shard>
+        </test_cluster2>
+        <test_cluster3>
+            <shard>
+                <internal_replication>true</internal_replication>
+                <replica>
+                    <host>node_1</host>
+                    <port>9000</port>
+                </replica>
+            </shard>
+        </test_cluster3>
+    </remote_servers>
+</yandex>
+'''
+
+
+def send_repeated_query(table, count=5):
+    for i in range(count):
+        node.query_and_get_error("SELECT count() FROM {} SETTINGS receive_timeout=1".format(table))
+
+
+def get_errors_count(cluster, host_name="node_1"):
+    return int(node.query("SELECT errors_count FROM system.clusters WHERE cluster='{}' and host_name='{}'".format(cluster, host_name)))
+
+
+def set_config(config):
+    node.replace_config("/etc/clickhouse-server/config.d/remote_servers.xml", config)
+    node.query("SYSTEM RELOAD CONFIG")
+
+
+def test_simple_reload(started_cluster):
+    send_repeated_query("distributed")
+
+    assert get_errors_count("test_cluster") > 0
+    
+    node.query("SYSTEM RELOAD CONFIG")
+
+    assert get_errors_count("test_cluster") > 0
+
+
+def test_update_one_cluster(started_cluster):
+    send_repeated_query("distributed")
+    send_repeated_query("distributed2")
+
+    assert get_errors_count("test_cluster") > 0
+    assert get_errors_count("test_cluster2") > 0
+
+    set_config(test_config1)
+
+    assert get_errors_count("test_cluster") == 0
```

Comment:
It works, because errors count is reset after update cluster config, right?

Reply:
Yes, here we change `test_cluster` config and it cause cluster reconstruction: delete old cluster with all it's information (including errors_count) and create new.